pipeline_dp/private_beam.py

```diff
@@ -0,0 +1,104 @@
+from apache_beam.transforms import ptransform
+from abc import abstractmethod
+from typing import Callable, Optional
+from apache_beam import pvalue
+
+import pipeline_dp
+from pipeline_dp import aggregate_params, budget_accounting
+
+
+class PrivateTransform(ptransform.PTransform):
+    """Abstract class for PrivateTransforms."""
+
+    def __init__(self, return_private: bool, label: Optional[str] = None):
+        super().__init__(label)
+        self._return_private = return_private
+        self._budget_accountant = None
+        self._privacy_id_extractor = None
+
+    def set_additional_parameters(
+            self, budget_accountant: budget_accounting.BudgetAccountant,
+            privacy_id_extractor: Callable):
+        """Set the additional parameters needed for the private transform."""
+        self._budget_accountant = budget_accountant
+        self._privacy_id_extractor = privacy_id_extractor
+
+    @abstractmethod
+    def expand(self, pcol: pvalue.PCollection) -> pvalue.PCollection:
+        pass
+
+
+class PrivateCollection:
+    """Private counterpart for PCollection."""
```

Comment:
Please add:

PrivateCollection guarantees that only anonymized data within the specified privacy budget can be extracted from it through PrivatePTransforms.